client/src/components/comments/index.tsx

```diff
@@ -0,0 +1,112 @@
+'use client';
+
+import { useState } from 'react';
+
+import { cn } from '@/utils';
+import { useUserInfo } from '@/context';
+import { UserIcon, Editor } from '@/components';
+
+import { IComment } from './types';
+import { Comment } from './comment';
+import { EditorBtnGroup } from './btn-group';
+
+export interface ICommentEditor {
+  placeholder: string;
+}
+
+export const CommentEditor = ({ placeholder }: ICommentEditor) => {
+  const [comments, setComments] = useState<IComment[]>([]);
+  const [activeComment, setActiveComment] = useState<string | null>(null);
+  const [isEditing, setIsEditing] = useState<string | null | false>(false);
+
+  const { _id, firstName, lastName, avatarUrl } = useUserInfo();
+
+  const authorId = _id ? String(_id) : undefined;
+
+  const onDisabled = () => {
```

Comment:
The onDisabled function is unnecessary since it only checks whether activeComment contains data. Instead, it's better to pass a boolean value isDisabled, which directly depends on activeComment !== null.